internal/pkg/cache/add.go

```diff
@@ -45,13 +47,30 @@ func (c *Cache) Add(opts *AddOpts) (cachedRegistry *Registry, err error) {
 	// to look up the source URL, but invalid packs metadata can mess that up.
 	// Throw error if registry source is not defined
 	if opts.Source == "" {
-		err = errors.ErrRegistrySourceRequired
-		return
+		err := errors.ErrRegistrySourceRequired
+		return cachedRegistry, err
 	}
 
-	cachedRegistry, err = c.addFromURI(opts)
+	cachedRegistry, err := c.addFromURI(opts)
+	if err != nil {
+		return cachedRegistry, err
+	}
 
-	return
+	if cachedRegistry != nil {
+		// Store a metadata JSON file for the cached registry
+		b, err := json.MarshalIndent(cachedRegistry, "", "  ")
+		if err != nil {
+			return cachedRegistry, err
+		}
+
+		metaPath := fmt.Sprintf("%s/%s/metadata.json", c.cfg.Path, opts.RegistryName)
+		err = os.WriteFile(metaPath, b, 0644)
+		if err != nil {
+			return cachedRegistry, err
+		}
```

Comment:
This could use less lines, although this is personal preference and not a required change.
```suggestion
		if err = os.WriteFile(metaPath, b, 0644); err != nil {
			return cachedRegistry, err
		}
```